airbyte-notification/src/main/java/io/airbyte/notification/SlackNotificationClient.java

```diff
@@ -95,7 +95,8 @@ public boolean notify(final String message) throws IOException, InterruptedExcep
         LOGGER.info("Successful notification ({}): {}", response.statusCode(), response.body());
         return true;
       } else {
-        LOGGER.warn("Failed to deliver notification ({}): {}", response.statusCode(), response.body());
+        final String errorMessage = String.format("Failed to deliver notification (%s): %s", response.statusCode(), response.body());
+        throw new IOException(errorMessage);
```

Comment:
Why did you decide to throw an exception?

Reply:
Throwing the exception here allows the API consuming this to handle the "NotificationRead" response as a fail or success.

Otherwise, the JobNotifier consumer is always try/catch exceptions too so it won't be passing through